### ISSUE TITLE
Running criugui failed

### ISSUE THREAD

**Issue by adyanthaya17** [ThomasJClark/criu-gui#1]
Hi,

I tried with this document but running the python command gives me the following error:

--> python -m criugui
Failed to connect to Mir: Failed to connect to server socket: No such file or directory
Unable to init server: Could not connect: Connection refused
Failed to connect to Mir: Failed to connect to server socket: No such file or directory
Unable to init server: Could not connect: Connection refused
/home/oss/criu-gui/criugui/__main__.py:19: PyGIWarning: Gtk was imported without specifying a version first. Use gi.require_version('Gtk', '3.0') before import to ensure that the right version gets loaded.
  from gi.repository import Gtk, GObject

(__main__.py:21591): Gtk-CRITICAL **: gtk_settings_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed
/usr/lib/python2.7/dist-packages/gi/overrides/__init__.py:326: Warning: g_object_get_qdata: assertion 'G_IS_OBJECT (object)' failed
  return super_init_func(self, **new_kwargs)
Segmentation fault

I have installed the gtk package:
--->dpkg -l libgtk2.0-0 libgtk-3-0

Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                      Version           Architecture      Description
+++-=========================-=================-================
ii  libgtk-3-0:amd64          3.18.9-1ubuntu3.3 amd64             GTK+ graphical user interface library
ii  libgtk2.0-0:amd64         2.24.30-1ubuntu1. amd64             GTK+ graphical user interface library

**Comment by ThomasJClark** [comment]
> Failed to connect to Mir: Failed to connect to server socket: No such file or directory

Mir is an Ubuntu thing right?  I don't see why this wouldn't work on Ubuntu but I've only ever tested it on RHEL and fedora.

Do you have problems running other GUI apps from the command line?  Maybe try some of the suggestions in [this thread](https://askubuntu.com/questions/871092/failed-to-connect-to-mir-failed-to-connect-to-server-socket-no-such-file-or-di)?

**Comment by adyanthaya17** [comment]
I tried with the link you mentioned and it seems like I am getting the same error:
Failed to connect to Mir: Failed to connect to server socket: No such file or directory
Unable to init server: Could not connect: Connection refused

It says in few links that MIR works after the gtk installation which has been on the Ubuntu machine.

**Comment by ThomasJClark** [comment]
I don't think the problem is GTK being installed, it sounds like there's some issue connecting to the display server.  Are you running a desktop environment on this machine?  Can you, for example, run `gedit` from the same terminal you're running this program?

**Comment by adyanthaya17** [comment]
This is what I get after running 'gedit': 

Failed to connect to Mir: Failed to connect to server socket: No such file or directory
Unable to init server: Could not connect: Connection refused

(gedit:8809): Gtk-WARNING **: cannot open display: :

I am running these commands on a Linux Ubuntu 16.04 host machine.

**Comment by ThomasJClark** [comment]
Yeah sounds like you don't have a display server running or you're running in a virtual terminal.  criu-gui is a desktop program and it requires a graphical environment to run

**Comment by adyanthaya17** [comment]
Then would it mean that it wouldn't work with an Ubuntu machine? Is there anyway we could get it running in the environment?

**Comment by ThomasJClark** [comment]
It should work on an Ubuntu machine if it's running a desktop environment.  Are you using Ubuntu Server?

**Comment by adyanthaya17** [comment]
Ah >.< The VM is a Ubuntu machine which running in the Remote Desktop Connection. I have been running the commands in the VM putty terminal.
